packages/fxa-content-server/server/lib/amplitude.js

```diff
@@ -101,6 +101,30 @@ const EVENTS = {
     group: GROUPS.rp,
     event: 'engage',
   },
+  'flow.email-domain-validation.engage': {
+    group: GROUPS.registration,
+    event: 'domain_validation_engage',
+  },
+  'flow.email-domain-validation.skip': {
+    group: GROUPS.registration,
+    event: 'domain_validation_skip',
+  },
+  'flow.email-domain-validation.success': {
+    group: GROUPS.registration,
+    event: 'domain_validation_success',
```

Comment:
It looks like `domain_validation_success`, `domain_validation_block` and `domain_validation_warn` should be changed to `domain_validation_result` with an event property `validation_result: (block|warn|success)` to help minimize the number of distinct event types. 

To do so, I *think* we'd have to:

1. Merge these into a single regexp based entry in [FUZZY_EVENTS](https://github.com/mozilla/fxa/blob/e9dbcf5dcb151f4d0ecbaa80db0a20a8bc4f82fd/packages/fxa-content-server/server/lib/amplitude.js#L120) like so:
```
[
  /^flow\.email\-domain\-validation\.(block|warn|success)$/, {
    group: GROUPS.registration,
    event: 'domain_validation_result'
 }
]
```

2. Add an [event property map function to fxa-shared](https://github.com/mozilla/fxa/blob/e9dbcf5dcb151f4d0ecbaa80db0a20a8bc4f82fd/packages/fxa-shared/metrics/amplitude.js#L51):

```js
function mapDomainValidationType(eventType, eventCategory, eventTarget) {
  // this function is called for all fxa_reg event types, only add the event properties
  // for the results pertaining to domain_validation_result.
  if (eventType === 'domain_validation_result') {
    return { validation_result: eventCategory };
  }
}
```

Really, I wish the event property mapper could be defined on the FUZZY_EVENT entry above, not sure how difficult it would be to add that, but it seems like that sort of logic should be as close as possible to the original entry, at least if it's not shared across multiple event types.

Reply:
@shane-tomlinson This commit was before @irrationalagent's comment.  Thanks for reviewing, and especially the tip on "fuzzy events".